contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -564,6 +566,15 @@ add_rewritten_query_fragment_to_mutator(PLtsql_expr_query_mutator *mutator)
 static void
 add_query_hints(PLtsql_expr *expr)
 {
+	ParserRuleContext* ctx = nullptr;
+	// If a query has both join hint and query hint which is a join hint, it should have all the join hints as the query hints as well
+	if (isJoinHintInOptionClause && ((join_hints_info[0] && !join_hints_info[3]) || (join_hints_info[1] && !join_hints_info[4]) || (join_hints_info[2] && !join_hints_info[5])))
```

Comment:
this is not readable as which hint each number represents, may be have a readable macro for each number to indicate the same #define LOOP_JOIN_HINT 0

Reply:
+1